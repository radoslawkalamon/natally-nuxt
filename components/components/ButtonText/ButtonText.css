.button-text {
  appearance: none;
  background-color: transparent;
  border: 0;
  box-shadow: none;
  color: inherit;
  cursor: pointer;
  font-family: var(--font-family-serif);
  font-size: inherit;
  padding: 0;
  text-decoration: underline;
}
