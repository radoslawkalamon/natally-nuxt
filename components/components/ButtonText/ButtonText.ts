import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsButtonText',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      required: true
    }
  }
})
