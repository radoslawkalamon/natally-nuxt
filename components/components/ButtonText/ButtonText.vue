<template>
  <button
    class="button-text"
    type="button"
    v-bind="$attrs"
    v-on="$listeners"
    v-text="label"
  />
</template>

<script src="./ButtonText.ts" lang="ts"></script>
<style src="./ButtonText.css" scoped></style>
