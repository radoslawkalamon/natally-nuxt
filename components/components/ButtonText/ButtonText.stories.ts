import { Meta, Story } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import ButtonText from '@/components/components/ButtonText/ButtonText.vue'

const meta: Meta = {
  title: 'Components / Button Text',
  component: ButtonText
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: { ButtonText },
  props: Object.keys(argTypes),
  template: '<ButtonText v-bind="$props" @click="onClick" />',
  methods: {
    onClick () {
      action('Click')()
    }
  }
})

export const Default: Story = Template.bind({})
Default.args = {
  label: 'Item label'
}
