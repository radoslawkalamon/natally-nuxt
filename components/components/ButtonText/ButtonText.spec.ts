import ButtonText from '@/components/components/ButtonText/ButtonText.vue'
import { shallHaveStringProp, shallRender } from '@/devtools/jest.shared.spec'

describe('Components / Button Text', () => {
  const defaultOptions = {
    propsData: {
      label: 'Item label'
    }
  }

  shallRender(ButtonText, defaultOptions)
  shallHaveStringProp(ButtonText, 'label', defaultOptions)
})
