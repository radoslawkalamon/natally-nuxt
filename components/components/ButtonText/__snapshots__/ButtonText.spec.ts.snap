// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Button Text Unit shall pass unit sanity test 1`] = `"<button type=\\"button\\" class=\\"button-text\\">Item label</button>"`;
